### PR TITLE
Wip add apidoc support

### LINKED ISSUE #12: Support browsing list of artifacts - not of default classifier (javadoc)
The apidoc server lets you browse through the list of available versions of an artifact through an URL-path like `<group-id>/<artifact-id>/`. However the links generated in the overview page always link to javadoc artifacts `<group-id>/<artifact-id>/<version>/javadoc/index.html` even if there are none. It would be very useful if this list contained links to all existing artifacts javadoc or non-javadoc.

### PR DESCRIPTION
**Pull request by seakayone**
fixes #12

### CODE REVIEW

**Comment by aconstantin** [comment]
will have a look next week. thanks for investing so much effort

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Tue Mar 03 15:16:01 CET 2015
+#Fri Feb 23 13:26:45 CET 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.5.1-bin.zip
```
Comment (aconstantin) [inline gradle/wrapper/gradle-wrapper.properties]: should be 4.8 or better 4.8.1

File: build.gradle [hunk 2]
```diff
@@ -84,8 +58,5 @@ bootRun {
         systemProperty 'spring.config.location', "file:${applicationProperties}"
     }
 }
-run {
-    if (project.hasProperty('applicationProperties')) {
-        systemProperty 'spring.config.location', "file:${applicationProperties}"
-    }
-}
+
+wrapper.gradleVersion = '4.8'
```
Comment (aconstantin) [inline build.gradle]: 4.8.1

File: src/main/java/com/rbmhtechnology/apidocserver/service/mavenrepo/MavenRepoClient.java [hunk 3]
```diff
@@ -1,94 +1,62 @@
...
-        new AuthScope(config.repositoryHost(), config.repositoryPort()),
-        new UsernamePasswordCredentials(credentials.username(), credentials.password())
-    ));
-    this.httpClient = HttpClients.custom().setDefaultCredentialsProvider(credsProvider).build();
+    this.httpWagon = new HttpWagon();
+    this.authenticationInfo = config.getCredentials().map(c -> {
+      final AuthenticationInfo authInfo = new AuthenticationInfo();
+      authInfo.setUserName(c.username());
+      authInfo.setPassword(c.password());
+      BasicAuthScope scope = new BasicAuthScope();
```
Comment (aconstantin) [inline src/main/java/com/rbmhtechnology/apidocserver/service/mavenrepo/MavenRepoClient.java]: those 4 lines, should be extracted from the mapping lambda

File: src/main/java/com/rbmhtechnology/apidocserver/controller/ApiDocController.java [hunk 4]
```diff
@@ -47,11 +49,18 @@
...
+  private final URL repositoryUrl;
 
   @Autowired
-  private GroupIdWhitelistService groupIdWhitelistService;
+  public ApiDocController(RepositoryService repositoryService,
+          GroupIdWhitelistService groupIdWhitelistService,
+          MavenRepositoryConfig mavenConfig) {
+    this.repositoryService = repositoryService;
+    this.groupIdWhitelistService = groupIdWhitelistService;
+    this. repositoryUrl = mavenConfig.repositoryUrl();
```
Comment (aconstantin) [inline src/main/java/com/rbmhtechnology/apidocserver/controller/ApiDocController.java]: formatting

File: readme.md [hunk 5]
```diff
@@ -26,7 +26,7 @@ The URL parts have following meaning:
 * `groupId` (mandatory): The group identifier of the artifact, e.g. `org.apache.commons`.
 * `artifactId` (mandatory): The artifact identifier, e.g. `commons-lang3`.
 * `version` (optional): The version identifier of the artifact, e.g. `3.3.2`. In case the version identifier is emitted, all available versions are listed.
-* `classifier` (optional): The documentation classifier to be shown. Typical values are `javadoc`, `scaladoc`, `groovydoc` and so on. If omitted the configurable default value `javadoc` is used instead.
+* `classifier` (optional): The documentation classifier to be shown. Typical values are `javadoc`, `scaladoc`, `groovydoc` and so on.
```
Comment (aconstantin) [inline readme.md]: describe behaviour if omitted

**Review by aconstantin — CHANGES_REQUESTED** [review]

**Comment by seakayone** [comment]
@aconstantin: Addressed your change requests for this pull request, please have another look.

**Review by aconstantin — APPROVED** [review]

**Comment by seakayone** [comment]
I have rebased the branch and clean up some commits as discussed offline, this is ready to be merged now.

**Pull request merged**